book/style_guide.md

```diff
@@ -0,0 +1,221 @@
+```
+
+#### One-line format
+
+One-line format is a format for writing all commands in one line.
+
+1. parameters:
+   1. **It's recommended to** put no spaces before and after pipe `|` symbol denoting block or closure parameter list beginning.
+   2. **It's recommended to** put one space after comma `,` after block or closure parameter parameter.
+   3. **It's recommended to** put no space before and one space after pipe `|` symbol denoting block or closure parameter list end.
+2. block and closure bodies:
+   1. **It's recommended to** put one space before closing block or closure curly brace `}`.
+3. records:
+   1. **It's recommended to** put one space before first record key and after last record key value.
+   2. **It's recommended to** put one space after `:` after record key.
+   3. **It's recommended to** put one space after comma `,` after key value.
+4. lists:
+   1. **It's recommended to** put no spaces before first list value and after last list value.
+   2. **It's recommended to** put one space after comma `,` after list value.
+5. surrounding constructs:
+   1. **It's recommended to** put one space before opening square `[` or curly brace `{` if preceding symbol is not the same.
+   2. **It's recommended to** put one space after closing square `]` or curly brace `}` if following symbol is not the same.
+   3. **It's recommended to** put no spaces between square `[]` or curly brackets `{}` with nothing between them.
+
+Correct:
+
+```nushell
+[[status]; [UP] [UP]] | all {|el| $el.status == UP }
+[1 2 3 4] | reduce {|it, acc| $it + $acc }
+{ x: 1, y: 2 }
+[1 2] | zip [3 4]
+[]
+```
+
+Incorrect:
+
+```nushell
+[[status]; [UP] [UP]] | all { |el| $el.status == UP } # two many spaces before "|el|": 2 instead of 0
+[1 2 3 4] | reduce {|it , acc | $it + $acc } # two many spaces after "it": 1 instead of 0
+                                             # two many spaces before "| $it + $acc": 1 instead of 0
+{x: 1, y : 2 } # two few spaces before "x: 1": 0 instead of 1
+               # two many spaces before ": 2": 1 instead of 0
+[1 2] | zip  [3 4] # two many spaces before "[3 4]": 2 instead of 1
+[ ] # two many spaces before "]": 1 instead of 0
+```
+
+#### Multi-line format
+
+Multi-line format is a format for writing all commands in several lines. It inherits all rules from one-line format
+and modifies them slightly.
+
+1. general:
+   1. **It's recommended to omit** trailing spaces.
+2. block and closure bodies:
+   1. **It's recommended to** put each body pipeline on a separate line.
+3. records:
+   1. **It's recommended to** put each record key-value pair on separate line.
+4. lists:
+   1. **It's recommended to** put each list item on separate line.
+5. surrounding constructs:
+   1. **It's recommended to** put one `\n` before opening square `[` or curly brace `{` if preceding symbol is not the same.
+   2. **It's recommended to** put one `\n` after closing square `]` or curly brace `}` if following symbol is not the same.
```

Comment:
when i read this, i understand a literal `\n` character, should we say _newline_ instead?

Reply:
I am not sure. I guess we can add this note explicitly:

> All backslash sequences should not be interpreted literally, unless it's told to do so. In other words,
treat something like `\n` like new line character and not literal slash followed by `n`.

Reply:
that might be confusing only to me :wink: